### PR TITLE
Add information about registration status to calendar

### LINKED ISSUE #1908: Clarify calendar colors
### Describe the bug
It's not always clear what the calendar colors mean for events.

### How to reproduce
Steps to reproduce the behaviour:
1. Go to calendar
2. See half-transparent event
3. Confused about whether it's unpublished/closed for registrations/hidden

### Expected behaviour
Some kind of help text to remind you what the colors mean. Maybe a legend or text on the event block in the calendar.

### Screenshots
Current state (the first two events will open registrations soon):
image

Example fix with text in the calendar block:

image

### PR DESCRIPTION
**Pull request by pingiun**
Closes #1908 

### Summary
Add information about registration status to calendar

### How to test
Steps to test the changes you made:
1. Go to calendar
2. See that events contain information about your registration status

### CODE REVIEW

**Comment by pingiun** [comment]
I also wanted to add hover text for the colored circle in the list view but the list view doesn't run the `eventRender` hook

File: website/events/static/events/js/main.js [hunk 1]
```diff
@@ -124,9 +124,15 @@ document.addEventListener('DOMContentLoaded', function () {
                 window.open(event.url, '_blank');
             }
         },
-        eventRender: function ({ el, event }) {
+        eventRender: function ({ el, event, isStart, view }) {
             el.setAttribute(
                 'title', event.extendedProps.description);
+            if (isStart && view.type !== 'list') {
+                let reg_info_elem = document.createElement('div');
```
Comment (se-bastiaan) [inline website/events/static/events/js/main.js:131]: Please use camelCase. Like the rest of this file.

Comment (se-bastiaan) [reply website/events/static/events/js/main.js:131]: This can also be a `const`

**Comment by se-bastiaan** [comment]
Isn't it possible to just add the extra text to to the title in the serializer with two newlines?

**Comment by pingiun** [comment]
> Isn't it possible to just add the extra text to to the title in the serializer with two newlines?

Don't think that's a very good idea tbh:

image

**Comment by se-bastiaan** [comment]
> > Isn't it possible to just add the extra text to to the title in the serializer with two newlines?
> 
> Don't think that's a very good idea tbh:
> 
> image

Ah didn't think about that

File: website/events/api/calendarjs/serializers.py [hunk 2]
```diff
@@ -26,6 +31,32 @@ def _class_names(self, instance):
...
 
+    def _registration_info(self, instance: Event):
+        if self.context["member"] and services.is_user_registered(
+            self.context["member"], instance
+        ):
+            queue_pos = services.user_registration_pending(
+                self.context["member"], instance
+            )
+            if isinstance(queue_pos, int):
+                return f"In waiting list at position {queue_pos}"
```
Comment (se-bastiaan) [inline website/events/api/calendarjs/serializers.py:42]: Can these strings get put into `_` for localisation, just to keep it consistent.

File: website/events/static/events/js/main.js [hunk 3]
```diff
@@ -124,9 +124,15 @@ document.addEventListener('DOMContentLoaded', function () {
                 window.open(event.url, '_blank');
             }
         },
-        eventRender: function ({ el, event }) {
+        eventRender: function ({ el, event, isStart, view }) {
             el.setAttribute(
                 'title', event.extendedProps.description);
+            if (isStart && view.type !== 'list') {
+                const regInfoElem = document.createElement('div');
+                regInfoElem.classList.add('mt-2');
```
Comment (se-bastiaan) [inline website/events/static/events/js/main.js:132]:
```suggestion
                regInfoElem.classList.add('mt-2', 'fc-title');
```

File: website/events/static/events/js/main.js [hunk 4]
```diff
@@ -124,9 +124,15 @@ document.addEventListener('DOMContentLoaded', function () {
...
+                regInfoElem.textContent = event.extendedProps.registration_info;
```
Comment (se-bastiaan) [inline website/events/static/events/js/main.js:133]:
```suggestion
                regInfoElem.title = event.extendedProps.registration_info;
                regInfoElem.textContent = event.extendedProps.registration_info;
```

image

**Comment by se-bastiaan** [comment]
Suggestion for the listview:

```
            eventIndicator.dataset.toggle = 'tooltip';
            eventIndicator.dataset.originalTitle = def.extendedProps.registration_info;
            eventIndicator.title = def.extendedProps.registration_info;
```

and `$('[data-toggle="tooltip"]').tooltip();` at the end of `renderEvents`

**Review by se-bastiaan — CHANGES_REQUESTED** [review]
See comments

**Comment by pingiun** [comment]
> Suggestion for the listview:
> 
> ```
>             eventIndicator.dataset.toggle = 'tooltip';
>             eventIndicator.dataset.originalTitle = def.extendedProps.registration_info;
>             eventIndicator.title = def.extendedProps.registration_info;
> ```
> 
> and `$('[data-toggle="tooltip"]').tooltip();` at the end of `renderEvents`

where should this be added?

**Comment by se-bastiaan** [comment]
> where should this be added?

The list view code. https://github.com/svthalia/concrexit/blob/master/website/events/static/events/js/listview.js#L50

**Review by se-bastiaan — APPROVED** [review]

**Comment by pingiun** [comment]
This has been reverted in 38.1 because of this:

image

**Pull request merged**